app/assets/javascripts/angular/controllers/posts/randomitem.js.erb

```diff
@@ -2,10 +2,21 @@ app.controller('RandomShowCtrl',['$scope', '$http', '$location', function($scope
 
   var url = "<%= ENV['URL'] %>" || "http://localhost:3000/";
 
+  $scope.post = {};
 
-  $http.get(url + 'posts').success(function(response){
-    $scope.post = response;
-    console.log($scope.post);
-  });
+  $scope.cross = function() {
+    newUrl = url + 'posts/random';
+
+    if ($scope.post.id !== undefined) {
+      newUrl + '/' + $scope.post.id;
+    }
+
+    $http.get(newUrl).success(function(response){
+      $scope.post = response;
+      console.log(response.title)
+    });
+  };
+
+  $scope.cross();
```

Comment:
Do you need to use the function `cross()` in the front end? If not, then you don't need to put it inside $scope